src/main/java/net/javadiscord/javabot/systems/staff_commands/RedeployCommand.java

```diff
@@ -39,8 +42,23 @@ public void execute(@NotNull SlashCommandInteractionEvent event) {
 			return;
 		}
 		log.warn("Redeploying... Requested by: " + event.getUser().getAsTag());
-		event.reply("**Redeploying...** This may take some time.").queue();
+		event.reply("Redeploying, this may take a few minutes.").queue();
+		try {
+			Process p = new ProcessBuilder("/bin/sh", Bot.getConfig().getSystems().getRedeployScriptLocation()).start();
+			p.waitFor();
+			String result = new String(p.getInputStream().readAllBytes());
+			if (result.contains("COMPILATION FAILED")) {
```

Comment:
It might be better to use the exit code of the script for checking whether it was successful or not.
For example, there could be errors other than the Gradle script failing (e.g. with `git clone`).

For failing with any error, `set -e` could be used in the script in order to abort the script as soon as any command fails.

Reply:
Yup, will do that soon™️